### PR TITLE
ApiTest. search entity from project space and shares jail

### PR DESCRIPTION
**Pull request by ScharfViktor**
Changes in 2 PR. related with https://github.com/owncloud/core/pull/40174
- added tests for searches resources in the project space and shared jail 
- added fail in expected failures https://github.com/owncloud/ocis/issues/4014

### CODE REVIEW

**Review by individual-it — COMMENTED** [review]
aren't those cases tested already in https://github.com/owncloud/core/blob/master/tests/acceptance/features/apiWebdavOperations/search.feature ?

**Comment by SagarGi** [comment]
> aren't those cases tested already in https://github.com/owncloud/core/blob/master/tests/acceptance/features/apiWebdavOperations/search.feature ?

@individual-it  I think `search.feature` has only tests for project space not for shares jail in core.

File: tests/acceptance/features/bootstrap/SpacesContext.php [hunk 1]
```diff
@@ -369,21 +371,20 @@ public function setUpScenario(BeforeScenarioScope $scope): void {
...
+	 *
 	 * @param  string $driveType
 	 *
 	 * @return void
 	 *
 	 * @throws Exception|GuzzleException
 	 */
-	public function deleteAllSpacesOfTheType(string $driveType): void
-	{
+	public function deleteAllSpacesOfTheType(string $driveType): void {
```
Comment (individual-it) [inline tests/acceptance/features/bootstrap/SpacesContext.php:387]: don't we run PHP lint here? this should have been caught by CI

**Review by individual-it — COMMENTED** [review]
LGTM, but before merging https://github.com/owncloud/core/pull/40174 needs to get merged and the `.drone.env` file reverted

**Comment by ScharfViktor** [comment]
> > aren't those cases tested already in https://github.com/owncloud/core/blob/master/tests/acceptance/features/apiWebdavOperations/search.feature ?
> 
> @individual-it I think `search.feature` has only tests for project space not for shares jail in core.

in core repository we have only tests for searching in personal space. so additionally I made a change in core

File: tests/acceptance/features/bootstrap/SpacesContext.php (hunk 1 above)
Comment (ScharfViktor) [reply tests/acceptance/features/bootstrap/SpacesContext.php:387]: I ran `make test-php-style-fix` and got `No fixable errors were found`

**Pull request merged**